server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/SignificantTermsAggregatorFactory.java

```diff
@@ -103,8 +103,10 @@ public Aggregator build(String name,
                                     Aggregator parent,
                                     SignificanceHeuristic significanceHeuristic,
                                     SignificantTermsAggregatorFactory sigTermsFactory,
+                                    boolean collectsFromSingleBucket,
                                     Map<String, Object> metadata) throws IOException {
 
+                assert collectsFromSingleBucket;
```

Comment:
Should we just throw an exception here?  I'm assuming relatively bad things happen if this condition is violated and the user will get garbage results (or a different exception)?

Reply:
Garbage, I think. It enforced by the code below and it mirrors what we do with `terms`. And I plan to drop it before release anyway, but I can certainly switch it. Because plans don't always work out.

Reply:
Ok, cool.  If it's enforced already no big deal :+1: